File: src/test/java/common/StaticAppender.java
package common;

import ch.qos.logback.classic.spi.ILoggingEvent;
import ch.qos.logback.core.AppenderBase;
import java.util.ArrayList;
import java.util.List;

public class StaticAppender extends AppenderBase<ILoggingEvent> {

  static List<ILoggingEvent> events = new ArrayList<>();

  @Override
  public void append(ILoggingEvent e) {
    events.add(e);
  }

  public static List<ILoggingEvent> getEvents() {
    return events;
  }

  public static void clearEvents() {
    events.clear();
  }

  public static ILoggingEvent getLastLoggedEvent() {
    return getEvents().get(0);
  }
}

File: src/test/kotlin/config/StaticAppender.kt
package config

import ch.qos.logback.classic.spi.ILoggingEvent
import ch.qos.logback.core.AppenderBase

class StaticAppender : AppenderBase<ILoggingEvent>() {
    public override fun append(e: ILoggingEvent) {
        events.add(e)
    }

    companion object {
        var events: MutableList<ILoggingEvent> = ArrayList()

        fun clearEvents() {
            events.clear()
        }
    }
}
